docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,518 @@
+  - type: Domain
+    value: httpbin.com
+    port: 80
+    protocol: http2
+  - type: Domain
+    value: httpbin.com
+    port: 443
+    protocol: tls
+ destination:
+   type: Passthrough
+```
+
+In this case, for port `80`, we require HTTP/2 configuration in the cluster settings. This necessitates having a separate configuration for each cluster.
+
+clusterName: `{policyName}_{port}`, so for the above policy name would looks like: `httpbin_80` and `httpbin_443`
+
+#### Extensability
+
+Provided model allows creating separate plugins which can integrate with different cloud providers or components. The user can create it's own custom configuration and later in the code create XDS configuration based on it. 
+
+Example:
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: example
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+spec:
+  match:
+  - type: InternalVIP
+    value: example.ext.svc.local
+    port: 80
+    protocol: http
+ destination:
+   type: Extension
+   extension:
+     type: Lambda 
+     config:
+       arn: arn:aws:lambda:us-west-2:123456789012:function:my-function
+```
+
+#### Universal without Transparent Proxy
+
+This specific setup has limitations caused by the nature of passthrough. 
+
+Limitations:
+* Wildcard doesn’t work through proxy
+* Only `destination.type: Manged` supported
+
+Traffic handled without transparent proxy doesn't provide information about the original destination of the request. That causes traffic to be difficult to recognize and we are not able to handle 2 different matches on the same listener, unless it's http traffic. In this case it makes sense to have only one match, or even use only destinations.
+
+#### IP intersections 
+
+It's possible for users to create a policy where the IP ranges intersect with each other. We cannot validate this during policy creation, but during configuration generation, we are able to check if IPs intersect. In such cases, we can log a message stating:
+
+> external-service1 and external-service2 have overlapping IPs X.X.X.X, which can disrupt your traffic.
+
+As a result, we will select the first `MeshExternalService` based on the creation type. If they were created simultaneously, we will use lexicographical order.
+
+#### Domain generation
+
+The `ExternalService` resource automatically allocated an internal IP each real domain (e.g. `httpbin.com`). While this approach offered benefits such as avoiding the allocation of a listener for a specific port listening on `0.0.0.0`, it obscured the existing domain behind a custom IP. With the introduction of `MeshExternalService`, we expose to users a type called `InternalVIP`, enabling the creation of domains for specific external services.
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: mongo
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+spec:
+  match:
+  - type: InternalVIP # Kuma will generate a domain
+    value: mongo.ext.svc.local
```

Comment:
Wait I thought the user wouldn't pick the hostname that this would be driven by hostNameGenerator

Reply:
There were 2 ideas:
* use hostname generator
* it makes sense to allow MeshOperator to define them inline. Since MeshOperator is responsible for both resource(`HostnameGenerator`, `MeshExternalService`) creation there should be no security threat.

@jakubdyszkiewicz WDYT?